src/sync.h

```diff
@@ -182,10 +182,10 @@ typedef CMutexLock<CCriticalSection> CCriticalBlock;
         (cs).lock();                                          \
     }
 
-#define LEAVE_CRITICAL_SECTION(cs) \
-    {                              \
-        (cs).unlock();             \
-        LeaveCritical();           \
+#define LEAVE_CRITICAL_SECTION(cs)   \
+    {                                \
+        (cs).unlock();               \
+        LeaveCritical((void*)(&cs)); \
```

Comment:
This should just say LeaveCritical(&(cs)). cs should be parenthesized since it's a macro argument, and the cast is unneeded.